metricbeat/module/http/server/server.go

```diff
@@ -83,10 +84,10 @@ func (m *MetricSet) Run(reporter mb.PushReporterV2) {
 				reporter.Error(err)
 			} else {
 				event := mb.Event{}
-				event.ModuleFields = common.MapStr{}
-				metricSetName := fields[mb.NamespaceKey].(string)
+				ns := fields[mb.NamespaceKey].(string)
 				delete(fields, mb.NamespaceKey)
-				event.ModuleFields.Put(metricSetName, fields)
+				event.MetricSetFields = fields
```

Comment:
Did you check if this has the expected outcome? Will this put it always under `server` or the namespace set below?

Reply:
it will put under the `http.<namespace>`